iam-password-policy/change-password-policy.py

```diff
@@ -0,0 +1,166 @@
+                logging.info("Existing Policy does not match auto generated policy. Changes would be applied.")
+                logging.info("Exiting")
+                return(False)
+        elif currentAccountPolicy is not inputPolicy:
+            logging.info("Existing Policy does not match user input policy. Changes would be applied.")
+            logging.info("Exiting")
+            return(False)
+        else:
+            logging.info(" Policies match no changes would be applied.")
+            logging.info("Exiting")
+            return(False)
+
+
+
+
+
+    if args.auto:
+        if currentAccountPolicy is not strong_stance:
+            response = client.update_account_password_policy(
+            MinimumPasswordLength=strong_stance["MinimumPasswordLength"],
+            RequireSymbols=strong_stance["RequireSymbols"],
+            RequireNumbers=strong_stance["RequireNumbers"],
+            RequireUppercaseCharacters=strong_stance["RequireUppercaseCharacters"],
+            RequireLowercaseCharacters=strong_stance["RequireLowercaseCharacters"],
+            AllowUsersToChangePassword=strong_stance["AllowUsersToChangePassword"],
+            MaxPasswordAge=strong_stance["MaxPasswordAge"],
+            PasswordReusePrevention=strong_stance["PasswordReusePrevention"],
+            HardExpiry=strong_stance["HardExpiry"],
+        )
+            changes_required = True
+
+    else:
+
+        if currentAccountPolicy is not inputPolicy:
+            response = client.update_account_password_policy(
+            MinimumPasswordLength=args.MinimumPasswordLength,
+            RequireSymbols=args.RequireSymbols,
+            RequireNumbers=args.RequireNumbers,
+            RequireUppercaseCharacters=args.RequireUppercaseCharacters,
+            RequireLowercaseCharacters=args.RequireLowercaseCharacters,
+            AllowUsersToChangePassword=args.AllowUsersToChangePassword,
+            MaxPasswordAge= args.MaxPasswordAge,
+            PasswordReusePrevention=args.PasswordReusePrevention,
+            HardExpiry=args.HardExpiry,
+            )
+            changes_required = True
+
+    if changes_required:
+        logging.info("Changes requested based on user input attempting update.")
+   
+    
+    if response['ResponseMetadata']['HTTPStatusCode'] == 200:
+        logging.info("Changes requested based on user input and were successfully applied.")
+
+        return(True)
+    else:
+        if not changes_required:
+            logging.info("Changes were not requested based on user input")
+        logging.info("Changes were not succesfully applied.")
+        return(False)
+        
+
+def do_args():
+    import argparse
+    parser = argparse.ArgumentParser()
+    parser.add_argument("--profile", help="Use this CLI profile (instead of default or env credentials)")
+    parser.add_argument("--auto", help="Use default 'strong' password policy stance")
+    parser.add_argument("--debug", help="Display debug logging messages")
```

Comment:
This should probably be 
`parser.add_argument("--debug", help="Display debug logging messages", action='store_true')`

Reply:
fixed